runtime/near-vm-logic/src/logic.rs

```diff
@@ -605,7 +535,12 @@ impl<'a> VMLogic<'a> {
     pub fn input(&mut self, register_id: u64) -> Result<()> {
         self.gas_counter.pay_base(base)?;
 
-        self.internal_write_register(register_id, self.context.input.clone())
+        self.registers.set(
+            &mut self.gas_counter,
+            &self.config.limit_config,
+            register_id,
+            self.context.input.clone(),
```

Comment:
```suggestion
            &self.context.input,
```

Reply:
Unfortunately because it’s a generic argument auto deref doesn’t happen and here we get ‘the trait `From<&Vec<u8>>` is not implemented for `Box<[u8]>`’ error. as_slice works here though.